presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -167,9 +173,34 @@ protected Boolean visitAtTimeZone(AtTimeZone node, Void context)
         @Override
         protected Boolean visitSubqueryExpression(SubqueryExpression node, Void context)
         {
+            /*
+             * Walk the subquery and check usages of our fields (the fields defined in our scope) that are not part of GROUP BY.
+             */
+            AstUtils.preOrder(node)
+                    .filter(Predicates.isInstanceOfAny(DereferenceExpression.class, QualifiedNameReference.class))
+                    .map(Expression.class::cast)
+                    .forEach(this::analyzeSubqueryReference);
+
             return true;
         }
 
+        private void analyzeSubqueryReference(Expression referenceExpression)
+        {
+            ResolvedField resolvedField = analysis.getScope(referenceExpression)
+                    .tryResolveField(referenceExpression)
+                    // Unresolvable fields should be found and reported earlier.
+                    .orElseThrow(() -> new IllegalStateException(format("Cannot resolve %s", referenceExpression)));
+
+            if (resolvedField.getScope() != scope) {
+                // Not our scope, not our responsibility
+                return;
+            }
+            if (!fieldIndexes.contains(resolvedField.getFieldIndex())) {
```

Comment:
I don't like this name `fieldIndexes`. It has no semantic information. What would you say to name it `groupingFields`?